vsphere/datadog_checks/vsphere/metrics.py

```diff
@@ -0,0 +1,631 @@
+# (C) Datadog, Inc. 2019-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+from pyVmomi import vim
+
+# https://code.vmware.com/apis/358/vsphere/doc/cpu_counters.html
+
+# Set of metrics that are emitted as percentages between 0 and 100. For those metrics, we divide the value by 100
+# to get a float between 0 and 1.
+PERCENT_METRICS = {
+    'cpu.capacity.contention.avg',
+    'cpu.coreUtilization.avg',
+    'cpu.coreUtilization.max',
+    'cpu.coreUtilization.min',
+    'cpu.coreUtilization.raw',
+    'cpu.corecount.contention.avg',
+    'cpu.demandEntitlementRatio.latest',
+    'cpu.latency.avg',
+    'cpu.readiness.avg',
+    'cpu.usage.avg',
+    'cpu.usage.max',
+    'cpu.usage.min',
+    'cpu.usage.raw',
+    'cpu.utilization.avg',
+    'cpu.utilization.max',
+    'cpu.utilization.min',
+    'cpu.utilization.raw',
+    'datastore.siocActiveTimePercentage.avg',
+    'disk.capacity.contention.avg',
+    'disk.scsiReservationCnflctsPct.avg',
+    'gpu.mem.usage.avg',
+    'gpu.mem.usage.max',
+    'gpu.mem.usage.min',
+    'gpu.mem.usage.raw',
+    'gpu.utilization.avg',
+    'gpu.utilization.max',
+    'gpu.utilization.min',
+    'gpu.utilization.raw',
+    'mem.capacity.contention.avg',
+    'mem.latency.avg',
+    'mem.reservedCapacityPct.avg',
+    'mem.usage.avg',
+    'mem.usage.max',
+    'mem.usage.min',
+    'mem.usage.raw',
+    'mem.vmfs.pbc.capMissRatio.latest',
+    'power.capacity.usagePct.avg',
+    'rescpu.actav1.latest',
+    'rescpu.actav15.latest',
+    'rescpu.actav5.latest',
+    'rescpu.actpk1.latest',
+    'rescpu.actpk15.latest',
+    'rescpu.actpk5.latest',
+    'rescpu.maxLimited1.latest',
+    'rescpu.maxLimited15.latest',
+    'rescpu.maxLimited5.latest',
+    'rescpu.runav1.latest',
+    'rescpu.runav15.latest',
+    'rescpu.runav5.latest',
+    'rescpu.runpk1.latest',
+    'rescpu.runpk15.latest',
+    'rescpu.runpk5.latest',
+    'storageAdapter.OIOsPct.avg',
+    'sys.diskUsage.latest',
+    'sys.resourceCpuAct1.latest',
+    'sys.resourceCpuAct5.latest',
+    'sys.resourceCpuMaxLimited1.latest',
+    'sys.resourceCpuMaxLimited5.latest',
+    'sys.resourceCpuRun1.latest',
+    'sys.resourceCpuRun5.latest',
+    'vcResources.priviledgedcpuusage.avg',
+    'vcResources.processcpuusage.avg',
+    'vcResources.systemcpuusage.avg',
+    'vcResources.systemnetusage.avg',
+    'vcResources.usercpuusage.avg',
+    'vsanDomObj.readCacheHitRate.latest',
+}
+
+# All metrics that can be collected from VirtualMachines.
+# The table maps a dd-formatted metric_name to a tuple containing:
+# (collection_level, per_instance_collection_level, (optional)is_available_per_instance)
```

Comment:
Where are `(collection_level, per_instance_collection_level, (optional)is_available_per_instance)` those values used? Can't find the usage 🤔

Reply:
Not used yet, but will definitely be useful in the future (like collecting the instance tag) :)
I spent enough time on this so that no one has to re-do that work :p

Reply:
In that case, maybe you can detail a bit more the inline doc and say it's not used for now, but can be used for other purposes by giving example(s). That will help even more if someone else will implement something using those information.

Reply:
👍

Reply:
Done